### PR TITLE
chore: Check logging functions like printf

### PR DESCRIPTION
**Pull request by srebhan**
Check all telegraf logger functions for the right number of arguments and the correct formatting directives. This enables the linter for those functions and fixes the fallout.

### CODE REVIEW

File: plugins/inputs/gnmi/handler.go [hunk 1]
```diff
@@ -137,7 +137,7 @@ func (h *handler) handleSubscribeResponseUpdate(acc telegraf.Accumulator, respon
 				// unmarshal extention
 				err := proto.Unmarshal(currentExt, juniperHeader)
 				if err != nil {
-					h.log.Errorf("unmarshal gnmi Juniper Header extention failed: %w", err)
+					h.log.Errorf("unmarshal gnmi Juniper Header extension failed: %v", err)
```
Comment (zak-pawel) [inline plugins/inputs/gnmi/handler.go:140]: If we are here, maybe we can start log message with capital letter?

File: plugins/inputs/gnmi/handler.go [hunk 2]
```diff
@@ -187,7 +187,7 @@ func (h *handler) handleSubscribeResponseUpdate(acc telegraf.Accumulator, respon
 			}
 			h.log.Debugf("Tag-subscription update for %q: %+v", tagSub.Name, update)
 			if err := h.tagStore.insert(tagSub, fullPath, fields, tags); err != nil {
-				h.log.Errorf("inserting tag failed: %w", err)
+				h.log.Errorf("inserting tag failed: %v", err)
```
Comment (zak-pawel) [inline plugins/inputs/gnmi/handler.go:190]: Start log message with capital letter?

File: plugins/inputs/intel_pmu/intel_pmu.go [hunk 3]
```diff
@@ -249,7 +249,7 @@ func (i *IntelPMU) Stop() {
 			}
 			err := event.Deactivate()
 			if err != nil {
-				i.Log.Warnf("failed to deactivate core event %q: %w", event, err)
+				i.Log.Warnf("failed to deactivate core event %q: %v", event, err)
```
Comment (zak-pawel) [inline plugins/inputs/intel_pmu/intel_pmu.go:252]: Start log message with capital letter?

File: plugins/inputs/intel_pmu/intel_pmu.go [hunk 4]
```diff
@@ -264,7 +264,7 @@ func (i *IntelPMU) Stop() {
 				}
 				err := event.Deactivate()
 				if err != nil {
-					i.Log.Warnf("failed to deactivate uncore event %q: %w", event, err)
+					i.Log.Warnf("failed to deactivate uncore event %q: %v", event, err)
```
Comment (zak-pawel) [inline plugins/inputs/intel_pmu/intel_pmu.go:267]: Start log message with capital letter?

File: plugins/inputs/intel_powerstat/intel_powerstat.go [hunk 5]
```diff
@@ -633,7 +633,7 @@ func (p *PowerStat) addCPUC6StateResidencyMetric(cpuID string, acc telegraf.Accu
 	coresData := p.msr.getCPUCoresData()
 	// Avoid division by 0
 	if coresData[cpuID].timeStampCounterDelta == 0 {
-		p.Log.Errorf("timestamp counter on offset %s should not equal 0 on cpuID %s",
+		p.Log.Errorf("timestamp counter on offset %d should not equal 0 on cpuID %s",
```
Comment (zak-pawel) [inline plugins/inputs/intel_powerstat/intel_powerstat.go:636]: Start log message with capital letter?

File: plugins/inputs/intel_powerstat/intel_powerstat.go [hunk 6]
```diff
@@ -657,7 +657,7 @@ func (p *PowerStat) addCPUC0StateResidencyMetric(cpuID string, acc telegraf.Accu
...
```
Comment (zak-pawel) [inline plugins/inputs/intel_powerstat/intel_powerstat.go:660]: Start log message with capital letter?

File: plugins/inputs/intel_powerstat/intel_powerstat.go [hunk 7]
```diff
@@ -734,7 +734,7 @@ func (p *PowerStat) getBusClock(cpuID string) float64 {
 		return p.getSilvermontBusClock(cpuID)
 	}
 
-	p.Log.Debugf("couldn't find the freq for the model: %d", model)
+	p.Log.Debugf("couldn't find the freq for the model: %s", model)
```
Comment (zak-pawel) [inline plugins/inputs/intel_powerstat/intel_powerstat.go:737]: Start log message with capital letter?

File: plugins/inputs/intel_powerstat/intel_powerstat.go [hunk 8]
```diff
@@ -748,7 +748,7 @@ func (p *PowerStat) getSilvermontBusClock(cpuID string) float64 {
 
 	i := int(msr & 0xf)
 	if i >= len(silvermontFreqTable) {
-		p.Log.Debugf("unknown msr value: %d, using default bus clock value: %d", i, silvermontFreqTable[3])
+		p.Log.Debugf("unknown msr value: %d, using default bus clock value: %f", i, silvermontFreqTable[3])
```
Comment (zak-pawel) [inline plugins/inputs/intel_powerstat/intel_powerstat.go:751]: Start log message with capital letter?

File: plugins/inputs/jti_openconfig_telemetry/jti_openconfig_telemetry.go [hunk 9]
```diff
@@ -438,7 +438,7 @@ func (m *OpenConfigTelemetry) Start(acc telegraf.Accumulator) error {
 
 		if m.Username != "" && m.Password != "" && m.ClientID != "" {
 			if err := m.authenticate(ctx, server, grpcClientConn); err != nil {
-				m.Log.Errorf("error authenticating to %s: %w", grpcServer, err)
+				m.Log.Errorf("error authenticating to %s: %v", grpcServer, err)
```
Comment (zak-pawel) [inline plugins/inputs/jti_openconfig_telemetry/jti_openconfig_telemetry.go:441]: Start log message with capital letter?

File: plugins/inputs/mongodb/mongodb.go [hunk 10]
```diff
@@ -156,7 +156,7 @@ func (m *MongoDB) Stop() {
 	for _, server := range m.clients {
 		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 		if err := server.client.Disconnect(ctx); err != nil {
-			m.Log.Errorf("disconnecting from %q failed: %s", server, err)
+			m.Log.Errorf("disconnecting from %q failed: %v", server.hostname, err)
```
Comment (zak-pawel) [inline plugins/inputs/mongodb/mongodb.go:159]: Start log message with capital letter?

File: plugins/inputs/opensearch_query/opensearch_query.go [hunk 11]
```diff
@@ -70,7 +70,7 @@ func (o *OpensearchQuery) Init() error {
 
 	err := o.newClient()
 	if err != nil {
-		o.Log.Errorf("error creating OpenSearch client: %w", err)
+		o.Log.Errorf("error creating OpenSearch client: %v", err)
```
Comment (zak-pawel) [inline plugins/inputs/opensearch_query/opensearch_query.go:73]: Start log message with capital letter?

File: plugins/inputs/upsd/upsd.go [hunk 12]
```diff
@@ -67,12 +67,12 @@ func (u *Upsd) gatherUps(acc telegraf.Accumulator, name string, variables []nut.
...
 	timeLeftS, err := internal.ToFloat64(metrics["battery.runtime"])
 	if err != nil {
-		u.Log.Warnf("'battery.runtime' type is not supported: %w", err)
+		u.Log.Warnf("'battery.runtime' type is not supported: %v", err)
 	}
 
 	timeLeftNS, err := internal.ToInt64(timeLeftS * 1_000_000_000)
 	if err != nil {
-		u.Log.Warnf("converting 'battery.runtime' to 'time_left_ns' failed: %w", err)
+		u.Log.Warnf("converting 'battery.runtime' to 'time_left_ns' failed: %v", err)
```
Comment (zak-pawel) [inline plugins/inputs/upsd/upsd.go:75]: Start log message with capital letter?

File: plugins/outputs/cloud_pubsub/cloud_pubsub.go [hunk 13]
```diff
@@ -189,7 +189,7 @@ func (ps *PubSub) toMessages(metrics []telegraf.Metric) ([]*pubsub.Message, erro
 
 		b, err = ps.compressData(b)
 		if err != nil {
-			ps.Log.Errorf("unable to compress message with %s: %w", ps.ContentEncoding, err)
+			ps.Log.Errorf("unable to compress message with %s: %v", ps.ContentEncoding, err)
```
Comment (zak-pawel) [inline plugins/outputs/cloud_pubsub/cloud_pubsub.go:192]: Start log message with capital letter?

File: plugins/outputs/mqtt/mqtt.go [hunk 14]
```diff
@@ -179,7 +179,7 @@ func (m *MQTT) collectNonBatch(hostname string, metrics []telegraf.Metric) []mes
 	for _, metric := range metrics {
 		topic, err := m.generator.Generate(hostname, metric)
 		if err != nil {
-			m.Log.Warnf("Generating topic name failed: %w", err)
+			m.Log.Warnf("Generating topic name failed: %v", err)
```
Comment (zak-pawel) [inline plugins/outputs/mqtt/mqtt.go:182]: Start log message with capital letter? (for line below)

File: plugins/outputs/mqtt/mqtt.go [hunk 15]
```diff
@@ -201,7 +201,7 @@ func (m *MQTT) collectBatch(hostname string, metrics []telegraf.Metric) []messag
...
```
Comment (zak-pawel) [inline plugins/outputs/mqtt/mqtt.go:204]: Start log message with capital letter? (for line below)

File: plugins/outputs/mqtt/mqtt.go [hunk 16]
```diff
@@ -225,7 +225,7 @@ func (m *MQTT) collectField(hostname string, metrics []telegraf.Metric) []messag
...
```
Comment (zak-pawel) [inline plugins/outputs/mqtt/mqtt.go:228]: Start log message with capital letter? (for line below)

File: plugins/outputs/mqtt/mqtt.go [hunk 17]
```diff
@@ -249,7 +249,7 @@ func (m *MQTT) collectHomieV4(hostname string, metrics []telegraf.Metric) []mess
...
```
Comment (zak-pawel) [inline plugins/outputs/mqtt/mqtt.go:252]: Start log message with capital letter? (for line below)

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go [hunk 18]
```diff
@@ -134,7 +134,7 @@ func (a *NebiusCloudMonitoring) Write(metrics []telegraf.Metric) error {
 		for _, field := range m.FieldList() {
 			value, err := internal.ToFloat64(field.Value)
 			if err != nil {
-				a.Log.Errorf("skipping value: %w", err.Error())
+				a.Log.Errorf("skipping value: %v", err)
```
Comment (zak-pawel) [inline plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go:137]: Start log message with capital letter?

File: plugins/outputs/opentelemetry/opentelemetry.go [hunk 19]
```diff
@@ -165,12 +165,12 @@ func (o *OpenTelemetry) sendBatch(metrics []telegraf.Metric) error {
 		case telegraf.Summary:
 			vType = common.InfluxMetricValueTypeSummary
 		default:
-			o.Log.Warnf("unrecognized metric type %Q", metric.Type())
+			o.Log.Warnf("unrecognized metric type %v", metric.Type())
```
Comment (zak-pawel) [inline plugins/outputs/opentelemetry/opentelemetry.go:168]: Start log message with capital letter?

File: plugins/outputs/opentelemetry/opentelemetry.go [hunk 20]
```diff
@@ -165,12 +165,12 @@ func (o *OpenTelemetry) sendBatch(metrics []telegraf.Metric) error {
...
 			continue
 		}
 		err := batch.AddPoint(metric.Name(), metric.Tags(), metric.Fields(), metric.Time(), vType)
 		if err != nil {
-			o.Log.Warnf("failed to add point: %s", err)
+			o.Log.Warnf("failed to add point: %v", err)
```
Comment (zak-pawel) [inline plugins/outputs/opentelemetry/opentelemetry.go:173]: Start log message with capital letter?

File: plugins/outputs/postgresql/table_manager.go [hunk 21]
```diff
@@ -89,7 +89,7 @@ func (tm *TableManager) MatchSource(ctx context.Context, db dbh, rowSource *Tabl
 			if isTempError(err) {
 				return err
 			}
-			tm.Postgresql.Logger.Errorf("permanent error updating schema for %s: %w", tagTable.name, err)
+			tm.Postgresql.Logger.Errorf("permanent error updating schema for %s: %v", tagTable.name, err)
```
Comment (zak-pawel) [inline plugins/outputs/postgresql/table_manager.go:92]: Start log message with capital letter?

File: plugins/outputs/postgresql/table_manager.go [hunk 22]
```diff
@@ -120,7 +120,7 @@ func (tm *TableManager) MatchSource(ctx context.Context, db dbh, rowSource *Tabl
 		if isTempError(err) {
 			return err
 		}
-		tm.Postgresql.Logger.Errorf("permanent error updating schema for %s: %w", metricTable.name, err)
+		tm.Postgresql.Logger.Errorf("permanent error updating schema for %s: %v", metricTable.name, err)
```
Comment (zak-pawel) [inline plugins/outputs/postgresql/table_manager.go:123]: Start log message with capital letter?

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go [hunk 23]
```diff
@@ -126,7 +126,7 @@ func (a *YandexCloudMonitoring) Write(metrics []telegraf.Metric) error {
 		for _, field := range m.FieldList() {
 			value, err := internal.ToFloat64(field.Value)
 			if err != nil {
-				a.Log.Errorf("skipping value: %w", err.Error())
+				a.Log.Errorf("skipping value: %v", err)
```
Comment (zak-pawel) [inline plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go:129]: Start log message with capital letter?

File: plugins/processors/scale/scale.go [hunk 24]
```diff
@@ -122,7 +123,7 @@ func (s *Scale) scaleValues(metric telegraf.Metric) {
 
 			v, err := internal.ToFloat64(field.Value)
 			if err != nil {
-				s.Log.Errorf("error converting %q to float: %w\n", field.Key, err)
+				s.Log.Errorf("error converting %q to float: %v", field.Key, err)
```
Comment (zak-pawel) [inline plugins/processors/scale/scale.go:126]: Start log message with capital letter?

File: plugins/serializers/cloudevents/cloudevents.go [hunk 25]
```diff
@@ -139,7 +139,7 @@ func (s *Serializer) batchEvents(metrics []telegraf.Metric) ([]byte, error) {
 	for _, m := range metrics {
 		e, err := s.createEvent(m)
 		if err != nil {
-			s.Log.Errorf("creating event for %v failed: %w", m, err)
+			s.Log.Errorf("creating event for %v failed: %v", m, err)
```
Comment (zak-pawel) [inline plugins/serializers/cloudevents/cloudevents.go:142]: Start log message with capital letter?

File: plugins/inputs/win_perf_counters/win_perf_counters.go [hunk 26]
```diff
@@ -452,7 +452,7 @@ func (m *WinPerfCounters) Gather(acc telegraf.Accumulator) error {
 			m.Log.Debugf("gathering from %s", hostInfo.computer)
 			start := time.Now()
 			err := m.gatherComputerCounters(hostInfo, acc)
-			m.Log.Debugf("gathering from %s finished in %.3fs", hostInfo.computer, time.Since(start))
+			m.Log.Debugf("gathering from %s finished in %v", hostInfo.computer, time.Since(start))
```
Comment (zak-pawel) [inline plugins/inputs/win_perf_counters/win_perf_counters.go:455]: Start log message with capital letter?

File: .golangci.yml [hunk 27]
```diff
@@ -152,6 +152,15 @@ linters-settings:
...
       G306: "0640"
+  govet:
+    settings:
+      ## Check the logging function like it would be a printf
+      printf:
+        funcs:
+          - (github.com/influxdata/telegraf.Logger).Debugf
+          - (github.com/influxdata/telegraf.Logger).Infof
+          - (github.com/influxdata/telegraf.Logger).Warnf
+          - (github.com/influxdata/telegraf.Logger).Errorf
```
Comment (zak-pawel) [inline .golangci.yml:163]: I believe it makes sense to add also:
```
          - (github.com/influxdata/telegraf.Logger).Debug
          - (github.com/influxdata/telegraf.Logger).Info
          - (github.com/influxdata/telegraf.Logger).Warn
          - (github.com/influxdata/telegraf.Logger).Error
```

These findings apears:
```
plugins/inputs/ctrlx_datalayer/ctrlx_datalayer.go:150:3    govet  printf: (github.com/influxdata/telegraf.Logger).Debug call has possible Printf formatting directive %q
plugins/inputs/kinesis_consumer/kinesis_consumer.go:223:6  govet  printf: (github.com/influxdata/telegraf.Logger).Debug call has possible Printf formatting directive %v
plugins/inputs/proxmox/proxmox.go:120:3                    govet  printf: (github.com/influxdata/telegraf.Logger).Error call has possible Printf formatting directive %v
plugins/outputs/mqtt/mqtt.go:170:4                         govet  printf: (github.com/influxdata/telegraf.Logger).Warn call has possible Printf formatting directive %s
plugins/outputs/wavefront/wavefront.go:91:3                govet  printf: (github.com/influxdata/telegraf.Logger).Debug call has possible Printf formatting directive %s
```

**Review by zak-pawel — CHANGES_REQUESTED** [review]

File: plugins/outputs/mqtt/mqtt.go (hunk 14 above)
Comment (srebhan) [reply plugins/outputs/mqtt/mqtt.go:182]: ???

File: plugins/outputs/mqtt/mqtt.go (hunk 15 above)
Comment (srebhan) [reply plugins/outputs/mqtt/mqtt.go:204]: ???

File: plugins/outputs/mqtt/mqtt.go (hunk 16 above)
Comment (srebhan) [reply plugins/outputs/mqtt/mqtt.go:228]: ???

**Review by zak-pawel — APPROVED** [review]
Perfect!

**Review by powersj — APPROVED** [review]

**Pull request merged**